modules/lang-groovy/src/test/java/org/elasticsearch/messy/tests/BucketSelectorTests.java

```diff
@@ -451,4 +463,47 @@ public void testPartiallyUnmapped() throws Exception {
             assertThat(field2SumValue + field3SumValue, greaterThan(100.0));
         }
     }
+
+    public void testEmptyBuckets() {
+        SearchResponse response = client().prepareSearch("idx_with_gaps")
+                .addAggregation(histogram("histo").field(FIELD_1_NAME).interval(1)
+                        .subAggregation(range("range").field(FIELD_2_NAME).addRange(1, 2).addRange(2, 3).addRange(3, 4)
+                                .subAggregation(having("bucket_selector").setBucketsPaths("_count").script(new Script("_value0 > 0")))))
```

Comment:
is it possible to write this test in a way that it either does not require a script, or uses a mock script?